### PR TITLE
[GCU] Add check to ensure operation is valid in RDMA validator

### PR DESCRIPTION
**Pull request by isabelmsft**
### Description of PR

Summary:
[GCU] Add check to ensure operation is valid in RDMA validator
ADO [25310959](https://msazure.visualstudio.com/One/_workitems/edit/25310959)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [x] 202305 

#### What is the motivation for this PR?
Ensure that operation is allowed in RDMA config update validator

### CODE REVIEW

**Comment by bingwang-ms** [comment]
The most tricky thing is, GCU will convert `add` into `replace` if the entry you are going to add is present.
So, even if `add` is not supported in the gcu validator, the operation can succeed because `add` is converted into `replace`, and `replace` is supported.
So we may add some logic to check if the entry has been there, and change operation from `add` to `replace` if present.

File: tests/generic_config_updater/test_pfcwd_status.py [hunk 1]
```diff
@@ -228,7 +228,7 @@ def test_start_pfcwd(duthost, extract_pfcwd_config, ensure_dut_readiness, stop_p
     try:
         tmpfile = generate_tmpfile(duthost)
         output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
-        if is_valid_platform_and_version(duthost, "PFC_WD", "PFCWD enable/disable"):
+        if is_valid_platform_and_version(duthost, "PFC_WD", "PFCWD enable/disable", "add"):
```
Comment (qiluo-msft) [inline tests/generic_config_updater/test_pfcwd_status.py:231]: >add [](http://example.com/codeflow?start=85&length=3)

If this `add` depends on above `op`, better to use variable and do not repeat constant strings.

Comment (isabelmsft) [reply tests/generic_config_updater/test_pfcwd_status.py]: This test case only tests 'add', I moved add to be in variable `op` which would help if we extend the test case

**Review by qiluo-msft — APPROVED** [review]
LGTM. Please check with other active reviewers.

**Review by bingwang-ms — APPROVED** [review]

**Comment by bingwang-ms** [comment]
LGTM. Also added the label for 202305 branch.

**Comment by bingwang-ms** [comment]
MSFT ADO 25264431

**Comment by StormLiangMS** [comment]
@isabelmsft cherry pick conflict, could you help to check?

**Comment by isabelmsft** [comment]
> @isabelmsft cherry pick conflict, could you help to check?
Opened PR https://github.com/sonic-net/sonic-mgmt/pull/10534 . Cherry-pick conflict was due to https://github.com/sonic-net/sonic-mgmt/pull/10214 missing from 202305. PR 10534 includes both necessary cherry-picks to 202305

**Pull request merged**